be/src/column/binary_column.cpp

```diff
@@ -66,24 +67,33 @@ void BinaryColumnBase<T>::append_selective(const Column& src, const uint32_t* in
     const auto& src_offsets = src_column.get_offset();
     const auto& src_bytes = src_column.get_bytes();
 
+    bool need_prefetch =
+            (size > SEQUENTIAL_DISTINCT * 2) && (src_offsets.back() > (1 << 22)) && src_offsets.size() > 64;
```

Comment:
maybe we need add a check for max src_offset. if offset large enough prefetch may may works as expected.

Reply:
here both check bytes size and num, the bytes size > 4 MB is large enough. you mean we should increase `src_offsets.size() > 64`?

Reply:
another question, why we need to check src_offsets.size()

Reply:
@silverbullet233, avoid src only contains several large strings. in the worst cases, src column just has a 4.1MB string.

Reply:
> @silverbullet233, avoid src only contains several large strings. in the worst cases, src column just has a 4.1MB string.

@fzhedu got it…better to add some comments here

Reply:
I mean if the src column is a very large column, it looks like it would cause cache pollution?